pkg/pquerier/querier.go

```diff
@@ -0,0 +1,245 @@
+package pquerier
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/nuclio/logger"
+	"github.com/pkg/errors"
+	"github.com/v3io/v3io-go-http"
+	"github.com/v3io/v3io-tsdb/internal/pkg/performance"
+	"github.com/v3io/v3io-tsdb/pkg/config"
+	"github.com/v3io/v3io-tsdb/pkg/partmgr"
+	"github.com/v3io/v3io-tsdb/pkg/utils"
+)
+
+// Create a new Querier interface
+func NewV3ioQuerier(container *v3io.Container, logger logger.Logger,
+	cfg *config.V3ioConfig, partMngr *partmgr.PartitionManager) *V3ioQuerier {
+	newQuerier := V3ioQuerier{
+		container: container,
+		logger:    logger.GetChild("Querier"),
+		cfg:       cfg,
+	}
+	newQuerier.partitionMngr = partMngr
+	newQuerier.performanceReporter = performance.ReporterInstanceFromConfig(cfg)
+	return &newQuerier
+}
+
+type V3ioQuerier struct {
+	logger              logger.Logger
+	container           *v3io.Container
+	cfg                 *config.V3ioConfig
+	partitionMngr       *partmgr.PartitionManager
+	performanceReporter *performance.MetricReporter
+}
+
+type SelectParams struct {
+	Name             string
+	Functions        string
+	From, To, Step   int64
+	Windows          []int
+	Filter           string
+	RequestedColumns []RequestedColumn
+
+	disableAllAggr    bool
+	disableClientAggr bool
+}
+
+func (s *SelectParams) getRequestedColumns() []RequestedColumn {
+	if s.RequestedColumns != nil {
+		return s.RequestedColumns
+	}
+	functions := strings.Split(s.Functions, ",")
```

Comment:
trim?

Reply:
In some part we want to automatically include the count aggregate. and in this case we do not want to. that's why it's wasn't reused. i can move the functions parsing to the `aggregate` pkg if it will make you feel better 😅

Reply:
Also the `RequestedColumn` still stores the function as string, so it's not really being parsed rather than just splitting the aggregates